src/StatusWriter.cxx

```diff
@@ -0,0 +1,144 @@
+#include <cstdio>
+
+#include "Status.hpp"
+#include "StatusWriter.hpp"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/writer.h"
+
+namespace FileWriter {
+namespace Status {
+
+void StdIOWriter::write(const StreamMasterStatus &data) {
+  std::cout << "stream_master :\t" << data.status << "\n";
+  for (size_t i = 0; i < data.topic.size(); ++i) {
+    std::cout << data.topic[i] << ":\n";
+    print(data.streamer_status[i]);
+    print(data.streamer_stats[i]);
+    std::cout << "\n";
+  }
+}
+
+void StdIOWriter::print(const StreamerStatusType &x) {
+  std::cout << "\tstatus :\n"
+            << "\t\tmessages :\t" << x.messages << "\n"
+            << "\t\tbytes :\t\t" << x.bytes << "\n"
+            << "\t\terrors :\t" << x.errors << "\n";
+}
+void StdIOWriter::print(const StreamerStatisticsType &x) {
+  std::cout << "\tstatistics :\n"
+            << "\t\tsize average :\t" << x.size_avg << "\n"
+            << "\t\tsize std :\t" << x.size_std << "\n"
+            << "\t\tfrequency average :\t" << x.freq_avg << "\n"
+            << "\t\tfrequency std :\t\t" << x.freq_std << "\n";
+}
+
+rapidjson::Document JSONWriter::write(const StreamMasterStatus &data) {
+  return write_impl(data);
+}
+
+rapidjson::Document JSONWriter::write_impl(const StreamMasterStatus &data) {
+  using namespace rapidjson;
+  Document d;
+  auto &a = d.GetAllocator();
+  d.SetObject();
+  { // message type
+    d.AddMember("type", "filewriter_streammaster_status", a);
+  }
+  { // stream master info
+    Value sm,ss;
+    sm.SetObject();
+    std::string s{Err2Str(StreamMasterError{data.status})};
+    ss.SetString(s.c_str(),s.size(),a);
```

Comment:
Workaround: I enabled ``RAPIDJSON_HAS_STDSTRING`` in the headerfile in order to use std::string directly